pkg/ccl/crosscluster/logical/logical_replication_job_test.go

```diff
@@ -489,6 +492,62 @@ func TestRandomTables(t *testing.T) {
 	compareReplicatedTables(t, s, "a", "b", tableName, runnerA, runnerB)
 }
 
+func TestRandomStream(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
+
+	eventCount := 1000
+	testState := struct {
+		syncutil.Mutex
+		count int
+		done  chan struct{}
+	}{
+		done: make(chan struct{}),
+	}
+
+	streamingKnobs := &sql.StreamingTestingKnobs{
+		DistSQLRetryPolicy: &retry.Options{
+			InitialBackoff: time.Microsecond,
+			MaxBackoff:     2 * time.Microsecond,
+		},
+		RunAfterReceivingEvent: func(_ context.Context) error {
+			testState.Lock()
+			defer testState.Unlock()
+			testState.count++
+			if testState.count == eventCount {
+				close(testState.done)
+			}
+			return nil
+		},
+	}
+	clusterArgs := testClusterBaseClusterArgs
+	clusterArgs.ServerArgs.Knobs.DistSQL = &execinfra.TestingKnobs{
+		StreamingTestingKnobs: streamingKnobs,
+	}
+	clusterArgs.ServerArgs.Knobs.Streaming = streamingKnobs
+
+	ctx := context.Background()
+	tc, s, runnerA, _ := setupLogicalTestServer(t, ctx, clusterArgs)
+	defer tc.Stopper().Stop(ctx)
+
+	desc := desctestutils.TestingGetPublicTableDescriptor(s.DB(), s.Codec(), "a", "tab")
+	// We use a low events per checkpoint since we want to
+	// generate errors at many different code locations.
+	uri := fmt.Sprintf("randomgen://?TABLE_ID=%d&SST_PROBABILITY=0&ERROR_PROBABILITY=0.10&EVENTS_PER_CHECKPOINT=5", desc.GetID())
+
+	streamStartStmt := "CREATE LOGICAL REPLICATION STREAM FROM TABLE tab ON $1 INTO TABLE tab"
+	runnerA.Exec(t, streamStartStmt, uri)
+
+	t.Logf("waiting for %d events", eventCount)
+	select {
+	case <-testState.done:
+	case <-time.After(45 * time.Second):
+		t.Logf("%s", string(allstacks.Get()))
+		t.Fatal("timed out waiting for events")
+	}
+
```

Comment:
for another day: we could run some validation on the LDR job. E.g: if fewer than numRetryable errors were sent, the stream is still running.

Reply:
one quick validation would be to assert that nothing is in the dlq?

Reply:
I'm not sure that is viable because there is not guarantee that our random stream doesn't produce conflicting KVs.